packages/shared/src/abstract_game.ts

```diff
@@ -15,6 +15,11 @@ export abstract class AbstractGame<GameConfig = object, GameState = object> {
     this.config = config ?? this.defaultConfig();
   }
 
+  /**
+   * Returns the SGF data for a game
+   */
+  abstract getSGF(): string;
```

Comment:
Let's make this optional, for three reasons:

1) Smaller PR (we can remove many `getSGF { return "" }`)
2) Future variant implementers won't be obligated to implement immediately (new variant creation is already pretty heavy)
3) We can hide the "Download" button for variants without a proper implementation

Optional functions can be implemented via [interface merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html):

```
export interface AbstractGame<GameConfig = object, GameState = object> {
  /**
   * Returns the SGF data for a game
   */
  getSGF?: () => string;
}

export class AbstractGame<GameConfig = object, GameState = object> {
     ... existing stuff ...
}
```